File: src/main/java/com/yerti/core/enchantmenets/CombatEnchant.java
package com.yerti.core.enchantmenets;

import org.bukkit.enchantments.Enchantment;
import org.bukkit.enchantments.EnchantmentTarget;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.inventory.ItemStack;

import java.util.function.Consumer;

public class CombatEnchant extends Enchantment {

    private String name;
    private int maxLevel;
    private int startLevel;
    private int id;
    private Consumer<EntityDamageByEntityEvent> consumer;


    /**
     * Creates a custom combat enchantment with an event on entity damage
     * @param name
     * @param maxLevel
     * @param startLevel
     * @param id
     * @param consumer
     */
    public CombatEnchant(String name, int maxLevel, int startLevel, int id, Consumer<EntityDamageByEntityEvent> consumer) {
        super(id);
        this.name = name;
        this.maxLevel = maxLevel;
        this.startLevel = startLevel;
        this.id = id;
        this.consumer = consumer;

    }


    @Override
    public String getName() {
        return name;
    }

    @Override
    public int getMaxLevel() {
        return maxLevel;
    }

    @Override
    public int getStartLevel() {
        return startLevel;
    }

    @Override
    public EnchantmentTarget getItemTarget() {
        return null;
    }

    @Override
    public boolean conflictsWith(Enchantment enchantment) {
        return false;
    }

    @Override
    public boolean canEnchantItem(ItemStack itemStack) {
        return false;
    }

    public void sendEvent(EntityDamageByEntityEvent t) {
        consumer.accept(t);
    }
}

File: src/main/java/com/yerti/core/enchantmenets/CombatEnchant.kt
package com.yerti.core.enchantmenets

import org.bukkit.enchantments.Enchantment
import org.bukkit.enchantments.EnchantmentTarget
import org.bukkit.event.entity.EntityDamageByEntityEvent
import org.bukkit.inventory.ItemStack
import java.util.function.Consumer

class CombatEnchant(private val name: String, private val maxLevel: Int, private val startLevel: Int, id: Int, private val consumer: Consumer<EntityDamageByEntityEvent>) : Enchantment(id) {
    override fun getName(): String {
        return name
    }

    override fun getMaxLevel(): Int {
        return maxLevel
    }

    override fun getStartLevel(): Int {
        return startLevel
    }

    override fun getItemTarget(): EnchantmentTarget? {
        return null
    }

    override fun conflictsWith(enchantment: Enchantment): Boolean {
        return false
    }

    override fun canEnchantItem(itemStack: ItemStack): Boolean {
        return false
    }

    fun sendEvent(t: EntityDamageByEntityEvent) {
        consumer.accept(t)
    }

}